src/soundio.c

```diff
@@ -33,36 +33,29 @@ static const enum SoundIoBackend available_backends[] = {
     SoundIoBackendDummy,
 };
 
-static int (*backend_init_fns[])(struct SoundIoPrivate *) = {
-    [SoundIoBackendNone] = NULL,
+typedef int (*backend_init_t)(struct SoundIoPrivate *);
+static backend_init_t *make_backend_init_fns()
```

Comment:
same thing here. let's have this be statically initialized without any runtime code, and we can ditch the c99 syntax.